flow/project.py

```diff
@@ -3489,16 +3879,12 @@ def hello(job):
         if name in cls._GROUP_NAMES:
             raise ValueError("A group with name '{}' is already registered.".format(name))
 
-        signature = inspect.signature(func)
-        for i, (k, v) in enumerate(signature.parameters.items()):
-            if i and v.default is inspect.Parameter.empty:
-                raise ValueError(
-                    "Only the first argument in an operation argument may not have "
-                    "a default value! ({})".format(name))
+        if not getattr(func, '_flow_aggregate', False):
+            func._flow_aggregate = aggregator.groupsof(1)
 
         # Append the name and function to the class registry
         cls._OPERATION_FUNCTIONS.append((name, func))
-        cls._GROUPS.append(FlowGroupEntry(name=name, options=""))
+        cls._GROUPS.append(FlowGroupEntry(name=name, options="", aggregator=func._flow_aggregate))
```

Comment:
To prevent an ordering requirement, we should check this for singleton groups at `FlowGroup` construction. Ultimately we decided that aggregates should be associated with operations so we shouldn't limit behavior now that will likely be removed.

Reply:
Noted, this will be tackled in a later PR

Reply:
Moved to #362.